app/main.py

```diff
@@ -1,3 +1,18 @@
-def cache(func):
-    # Write your code here
-    pass
+def cache(func: callable) -> callable:
+    # dictionary 'archive' will store arguments given to func (as keys)
```

Comment:
it's not a good practice to write comments for your code a lot. code should be self-descriptive

Reply:
And actually, it is self-descriptive in this function

Reply:
you can use docstring to explain what the function does